homeassistant/components/zwave_js/sensor.py

```diff
@@ -295,3 +310,68 @@ def extra_state_attributes(self) -> dict[str, str] | None:
             return None
         # add the value's int value as property for multi-value (list) items
         return {"value": self.info.primary_value.value}
+
+
+class ZWaveNodeStatusSensor(SensorEntity):
+    """Representation of a node status sensor."""
+
+    def __init__(
+        self, config_entry: ConfigEntry, client: ZwaveClient, node: ZwaveNode
+    ) -> None:
+        """Initialize a generic Z-Wave device entity."""
+        self.config_entry = config_entry
+        self.client = client
+        self.node = node
+        name: str = (
+            self.node.name
+            or self.node.device_config.description
+            or f"Node {self.node.node_id}"
+        )
+        # Entity class attributes
+        self._attr_name = f"{name}: Node Status"
+        self._attr_unique_id = (
+            f"{self.client.driver.controller.home_id}.{node.node_id}.node_status"
+        )
+        # device is precreated in main handler
+        self._attr_device_info = {
+            "identifiers": {get_device_id(self.client, self.node)},
+        }
+        self._attr_state: str = node.status.name.lower()
+
+    async def async_poll_value(self, _: bool) -> None:
+        """Poll a value."""
+        raise ValueError("There is no value to poll for this entity")
+
+    def _status_changed(self, _: dict) -> None:
+        """Call when status event is received."""
+        self._attr_state = self.node.status.name.lower()
+        self.async_write_ha_state()
+
+    async def async_added_to_hass(self) -> None:
+        """Call when entity is added."""
+        # Add value_changed callbacks.
+        for evt in ("wake up", "sleep", "dead", "alive"):
+            self.async_on_remove(self.node.on(evt, self._status_changed))
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                f"{DOMAIN}_{self.unique_id}_poll_value",
+                self.async_poll_value,
+            )
+        )
+        self.async_write_ha_state()
+
+    @property
+    def available(self) -> bool:
+        """Return entity availability."""
+        return self.client.connected and bool(self.node.ready)
+
+    @property
+    def should_poll(self) -> bool:
```

Comment:
Why did we re-add these properties? I asked to put them as class attributes.

Reply:
Doh misread. Will submit a PR later today